blockstack_client/config.py

```diff
@@ -313,403 +317,406 @@ def get_logger( debug=DEBUG ):
 
-def opt_restore( prefix, opts ):
-   """
-   Given a dict of opts, add the given prefix to each key
-   """
+    return ret
 
-   ret = {}
 
-   for (opt_name, opt_value) in opts.items():
+def opt_restore(prefix, opts):
+    """
+    Given a dict of opts, add the given prefix to each key
+    """
 
-      ret[ prefix + opt_name ] = opt_value
+    return {prefix + name: value for name, value in opts.items()}
 
-   return ret
 
+def default_bitcoind_opts(config_file=None, prefix=False):
+    """
+    Get our default bitcoind options, such as from a config file,
+    or from sane defaults
+    """
 
-def default_bitcoind_opts( config_file=None, prefix=False ):
-   """
-   Get our default bitcoind options, such as from a config file,
-   or from sane defaults
-   """
+    default_bitcoin_opts = virtualchain.get_bitcoind_config(config_file=config_file)
 
-   default_bitcoin_opts = virtualchain.get_bitcoind_config( config_file=config_file )
-   
-   # strip None's
-   for (k, v) in default_bitcoin_opts.items():
-      if v is None:
-         del default_bitcoin_opts[k]
+    # drop dict values that are None
+    default_bitcoin_opts = {k: v for k, v in default_bitcoin_opts.items() if v is not None}
 
-   # strip 'bitcoind_'
-   if not prefix:
-       default_bitcoin_opts = opt_strip("bitcoind_", default_bitcoin_opts)
+    # strip 'bitcoind_'
+    if not prefix:
+        default_bitcoin_opts = opt_strip('bitcoind_', default_bitcoin_opts)
 
-   return default_bitcoin_opts
+    return default_bitcoin_opts
 
 
-def client_uuid_path( config_dir=CONFIG_DIR ):
+def client_uuid_path(config_dir=CONFIG_DIR):
     """
     where is the client UUID stored
     """
-    uuid_path = os.path.join(config_dir, "client.uuid")
+    uuid_path = os.path.join(config_dir, 'client.uuid')
     return uuid_path
 
 
-def get_or_set_uuid( config_dir=CONFIG_DIR ):
+def get_or_set_uuid(config_dir=CONFIG_DIR):
     """
     Get or set the UUID for this installation.
     Return the UUID either way
     Return None on failure
     """
     uuid_path = client_uuid_path(config_dir=config_dir)
     u = None
-    if not os.path.exists(uuid_path):
-       try:
-           u = str(uuid.uuid4())
-           with open(uuid_path, "w") as f:
-               f.write(u)
-               f.flush()
-               os.fsync(f.fileno())
-
-       except Exception, e:
+    if os.path.exists(uuid_path):
+        try:
+            with open(uuid_path, 'r') as f:
+                u = f.read()
+                u = u.strip()
+        except Exception as e:
             log.exception(e)
             return None
+    else:
+        try:
+            u = str(uuid.uuid4())
+            with open(uuid_path, 'w') as f:
+                f.write(u)
+                f.flush()
+                os.fsync(f.fileno())
+        except Exception as e:
+            log.exception(e)
+            return None
+
+    return u
+
+
+def configure(config_file=CONFIG_PATH, force=False, interactive=True):
```

Comment:
Note to myself--the interactive variant of this method is not covered by the integration tests.  Changes here will either need to be tested by hand before merging, or we'll need an `expect` script handy.